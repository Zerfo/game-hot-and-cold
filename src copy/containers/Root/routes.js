import Start from 'pages/Start';
import History from 'pages/History';
import Game from 'pages/Game';

const routes = [
  {
    exact: true,
    component: Start,
    path: '/',
  },
  {
    component: History,
    path: '/history',
  },
  {
    component: Game,
    path: '/game',
  }
];

export default routes;
