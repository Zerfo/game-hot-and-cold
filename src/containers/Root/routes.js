import Start from 'pages/Start';
import History from 'pages/History';
import Game from 'pages/Game';
import Instruction from 'pages/Instruction';

const routes = [
  {
    exact: true,
    component: Start,
    path: '/',
  },
  {
    component: History,
    path: '/history',
  },
  {
    component: Game,
    path: '/game',
  },
  {
    component: Instruction,
    path: 'instruction',
  }
];

export default routes;
