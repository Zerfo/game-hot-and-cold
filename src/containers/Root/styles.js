import { createStyles, makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() =>
  createStyles({
    container: {
      padding: 0,
      // width: '100vw',
      // height: '100vh',
    }
  }),
);

export default useStyles;
