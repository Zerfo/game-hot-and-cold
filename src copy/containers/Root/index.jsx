import React from 'react';

import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { Container } from '@material-ui/core';

import Header from 'components/Header';

import history from 'utils/history';

import routes from './routes';
import useStyles from './styles';

function Root() {
  const classes = useStyles();

  return (
    <Container className={classes.container}>
      <Router history={history}>
        <Header />
        <Switch>
          {
            routes.map((route, index) => <Route key={index} {...route} />)
          }
        </Switch>
      </Router>
    </Container>
  )
};

export default Root;
