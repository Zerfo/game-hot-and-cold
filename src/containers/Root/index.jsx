import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import { Container } from '@material-ui/core';

import Header from 'components/Header';

import store from 'store';

import routes from './routes';
import useStyles from './styles';

function Root() {
  const classes = useStyles();

  return (
    <Provider store={store}>
      <Container className={classes.container}>
        <Router>
          <Header />
          <Switch>
            {routes.map((route, index) => (
              <Route key={index} {...route} />
            ))}
          </Switch>
        </Router>
      </Container>
    </Provider>
  );
}

export default Root;
