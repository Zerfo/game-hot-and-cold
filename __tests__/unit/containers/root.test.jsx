import React from 'react';
import { shallow } from 'enzyme';
import Root from '../../../src/containers/Root';

it('renderers without crashing', () => {
  const component = shallow(<Root />);

  expect(component).toMatchSnapshot();
});
