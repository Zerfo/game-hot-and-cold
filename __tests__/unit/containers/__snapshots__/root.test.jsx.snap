// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderers without crashing 1`] = `ShallowWrapper {}`;
